pkg/ccl/backupccl/file_sst_sink.go

```diff
@@ -268,32 +268,13 @@ func (s *fileSSTSink) write(ctx context.Context, resp exportedSpan) error {
 
 	log.VEventf(ctx, 2, "writing %s to backup file %s", span, s.outName)
 
-	// Copy SST content.
-	sst, err := storage.NewMemSSTIterator(resp.dataSST, false)
-	if err != nil {
+	// To speed up SST reading, surface all the point keys first, flush,
+	// then surface all the range keys and flush.
+	if err := s.copyPointKeys(resp.dataSST); err != nil {
```

Comment:
It makes me a little sad we have to open up an iterator on this twice, and read the footer, the index, etc again just to change the iter opts.

I wonder if it'd make sense to include "OnlyPointKeys" in the ExportResponse or something the client could use as an optimization?

Reply:
We have the infrastructure in place to reconfigure an iterator, just need to expose the API for it. Want me to submit a PR?

Reply:
wait is the idea to add `KeyTypes:   storage.IterNonInterleavedPointAndRangeKeys,` i.e. a single iterator will surface all point keys and then all range keys in the given span? That seems great to me.

I'm not sure if ExportRequest needs to change, but I'll leave that decision to Erik.

Reply:
Although I'm unsure how big a perf improvement we get. I'm going to benchmark the current PR, as is, to understand how big of a regression there is by introducing these two iterators here.

Reply:
Nah, I think David's idea is for the `ExportResponse` to contain information about whether the SST contains point and/or range keys, and my suggestion is to export `MVCCIterator.SetOptions()` so you can do `iter.SetOption(IterOptions{KeyTypes: IterKeyTypeRangesOnly}))`.

Reply:
aha, makes sense! That seems like a worthwhile quick win -- especially for backups without revision history, which will never export range keys.

Reply:
Yeah, exactly, if we somehow know we didn't export a range key, we could just skip re-opening/reconfiguring the iterator entirely.

Reply:
It's possible that we're already tracking that metadata in the SST itself though. Can we get that from the SST reader @jbowens?

Reply:
Whether or not the table has a range key? Yeah, there's a sstable recording the count. We can test `reader.Properties.NumRangeKeys() > 0`